File: app/src/main/java/com/example/deimos/fwp/Product.java
package com.example.deimos.fwp;

import android.os.Parcel;
import android.os.Parcelable;

public class Product implements Parcelable {
    public final String name;
    public final String info;
    public final String tel;
    public final String email;

    public Product(String name,String info,String tel,String email) {
        this.name = name;
        this.info = info;
        this.tel = tel;
        this.email = email;

    }


    protected Product(Parcel in) {
        name = in.readString();
        info = in.readString();
        tel = in.readString();
        email = in.readString();
    }

    public static final Creator<Product> CREATOR = new Creator<Product>() {
        @Override
        public Product createFromParcel(Parcel in) {
            return new Product(in);
        }

        @Override
        public Product[] newArray(int size) {
            return new Product[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(name);
    }
}

File: app/src/main/java/com/example/deimos/fwp/Product.kt
package com.example.deimos.fwp

import android.os.Parcel
import android.os.Parcelable

class Product : Parcelable {
    override fun writeToParcel(dest: Parcel?, flags: Int) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    val name: String?
    val info: String?
    val tel: String?
    val email: String?
    val lat: String?
    val lng: String?
    val id: String?
    constructor(name: String, info: String, tel: String, email: String, lat: String, lng: String, id :String) {
        this.name = name
        this.info = info
        this.tel = tel
        this.email = email
        this.lat = lat
        this.id = id
        this.lng = lng

    }


    protected constructor(`in`: Parcel) {
        name = `in`.readString()
        info = `in`.readString()
        tel = `in`.readString()
        email = `in`.readString()
        lat = `in`.readString()
        lng = `in`.readString()
        id = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Product> {
        override fun createFromParcel(parcel: Parcel): Product {
            return Product(parcel)
        }

        override fun newArray(size: Int): Array<Product?> {
            return arrayOfNulls(size)
        }
    }


}
